### PR TITLE
Session continuation

### PR DESCRIPTION
**Pull request by czue**
(PR into https://github.com/dimagi/commcare-hq/pull/7485)

### CODE REVIEW

File: corehq/db.py [hunk 1]
```diff
@@ -59,6 +59,11 @@ def get_engine(self, engine_id=DEFAULT_ENGINE_ID):
         """
         return self._get_or_create_factory(engine_id).engine
 
+
+    def close_scoped_sessions(self):
```
Comment (dimagimon) [inline corehq/db.py]: E303 too many blank lines (2)

**Comment by czue** [comment]
@snopoke

File: corehq/apps/userreports/sql/adapter.py [hunk 2]
```diff
@@ -31,6 +32,12 @@ def drop_table(self):
         with self.engine.begin() as connection:
             self.get_table().drop(connection, checkfirst=True)
 
+    def get_query_object(self):
+        """
+        Get a sqlalchemy query object ready to query this table
+        """
+        return self.sesssion_factory.Session.query(self.get_table())
```
Comment (snopoke) [inline corehq/apps/userreports/sql/adapter.py]: I think `.query` should be called on an actual `Session` object and not the session maker so this should be `self.sesssion_factory.Session().query(self.get_table())`

Comment (snopoke) [inline corehq/apps/userreports/sql/adapter.py]: Never mind, I see you can do it this way with scoped sessions: http://docs.sqlalchemy.org/en/rel_1_0/orm/contextual.html#implicit-method-access

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**